src/nbl/asset/utils/CIESProfile.h

```diff
@@ -60,15 +64,46 @@ namespace nbl
                 IES_STORAGE_FORMAT getMaxValue() const { return maxValue; }
 
                 const IES_STORAGE_FORMAT& getIntegralFromGrid() const { return integral; }
+                const IES_STORAGE_FORMAT& getAvgEmmision() const { return avgEmmision; }
 
-                core::smart_refctd_ptr<asset::ICPUImageView> createCDCTexture(const size_t& width = CDC_DEFAULT_TEXTURE_WIDTH, const size_t& height = CDC_DEFAULT_TEXTURE_HEIGHT) const;
+                template<class ExecutionPolicy>
+                core::smart_refctd_ptr<asset::ICPUImageView> createIESTexture(ExecutionPolicy&& policy, const size_t& width = CDC_DEFAULT_TEXTURE_WIDTH, const size_t& height = CDC_DEFAULT_TEXTURE_HEIGHT) const;
+                core::smart_refctd_ptr<asset::ICPUImageView> createIESTexture(const size_t& width = CDC_DEFAULT_TEXTURE_WIDTH, const size_t& height = CDC_DEFAULT_TEXTURE_HEIGHT) const;
+                
+                template<class ExecutionPolicy>
+                static bool flattenIESTexture(ExecutionPolicy&& policy, const IES_STORAGE_FORMAT& avgEmmision, core::smart_refctd_ptr<asset::ICPUImageView> inoutIES, const float flatten = 0.0);
+                static bool flattenIESTexture(const IES_STORAGE_FORMAT& avgEmmision, core::smart_refctd_ptr<asset::ICPUImageView> inoutIES, const float flatten = 0.0);
```

Comment:
this needs to go into the Mitsuba Loader / Material Frontend, where the IES profiles get cached and loaded

Reply:
ok there's one possibility, this could be a non-static method, then it could be merged with `createIESTexture` with `flatten` being the last parameter modifying the `sample` directly.

But for this you need the Max, Integral, etc. to be computed with parsing so all the values are ready.

Reply:
I would keep separate `flattenIESTexture` while still allow to pass flatten into `createIESTexture` because it wouldn't be nice to resample just to flatten an ies, this way the utility can blend ready texels values from `inoutIES` but it's possible to flatten ies being created in place

Reply:
meh I will move it somewhere to Mitsuba as suggested, we don't have access to IES profile directly anyway but to meta only

Reply:
you'll probably never use both. 

If you want to sample only once because you'll flatten, you can load with `ELPF_LOAD_METADATA_ONLY` and `createIESTexture` with correct flatten.

I think sampling again and blending might be faster than decoding from R16_UNORM, because all `hAngles` and `vAngles` fit in cache and `data` is much sparser than your already sampled texture

Reply:
sure, so let's merge it to `createIESTexture`